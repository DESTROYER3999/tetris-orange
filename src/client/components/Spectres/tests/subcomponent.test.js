import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SpectresSub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const spectres = [
  {
    board: [0, 0, 0],
    playerName: "Player"
  }
];

describe("<LoginSub />", () => {
  test("SpectresSub rendering with no spectres", () => {
    const wrapper = shallow(<SpectresSub spectres={spectres} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
