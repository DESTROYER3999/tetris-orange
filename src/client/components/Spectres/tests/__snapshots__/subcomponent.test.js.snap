// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> SpectresSub rendering with no spectres 1`] = `
<styled.div>
  <div
    key="0"
  >
    <BoardSub
      board={
        Array [
          0,
          0,
          0,
        ]
      }
      isSpectre={true}
      playerName="Player"
    />
    <p
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Player
    </p>
  </div>
</styled.div>
`;
